[TensorExpr] Docs: checkin 'Conditionals in TE' doc.

ZolotukhinM opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#56949 [TensorExpr] Docs: checkin 'Conditionals in TE' doc.**

Differential Revision: [D28009077](https://our.internmc.facebook.com/intern/diff/D28009077)

Commit:
[TensorExpr] Docs: checkin 'Conditionals in TE' doc.

[ghstack-poisoned]

Commit:
Update on "[TensorExpr] Docs: checkin 'Conditionals in TE' doc."




[ghstack-poisoned]

navahgar commented on torch/csrc/jit/tensorexpr/ConditionalsInTE.md:
```diff
@@ -0,0 +1,40 @@
+
+It's clear that the fused approach is strictly more expressive than the unfused approach (each of the unfused operators can be implemented with the fused operator).  Thus the criteria for selecting between must be driven by either performance or ease of use.
+
+On the performance side, the fused approach can make a meaningful difference in performance when dealing with CPU vectors.  In the unfused approach the representation of a boolean is baked into the IR as either 0 or 1, since the comparison operation has to produce it for the selection operation to consume it.  In many vector ISAs, though, per-lane booleans are better represented as 0 or -1, or by ...
```
s/effective/effectively/

navahgar commented on torch/csrc/jit/tensorexpr/ConditionalsInTE.md:
```diff
@@ -0,0 +1,40 @@
+
+## Lazy vs Eager Conditionals
+
+Within the space of expression conditionals, Tensor IR contains a distinction between eager conditionals (`CompareSelect`) and lazy (or short-circuit) conditionals (`IfThenElse`).  The important distinction between these two is when the selectands (the `C` and `D` sub-expressions in the operation `(A cmp B) ? C : D` ).  In eager conditionals, all of the operands are allowed to be evaluated before ...
```
This sentence is incomplete.

navahgar reviewed (approved):
LGTM. Couple of minor comments below.

Commit:
Update on "[TensorExpr] Docs: checkin 'Conditionals in TE' doc."


Differential Revision: [D28009077](https://our.internmc.facebook.com/intern/diff/D28009077)

[ghstack-poisoned]
